### ISSUE TITLE
API documentation: which `service_name` values are supported?

### ISSUE THREAD

**Issue by fingolfin** [lemurheavy/coveralls-public#1243]
On <https://docs.coveralls.io/supported-ci-services> you list a bunch of CI services you support. On <https://docs.coveralls.io/api-introduction> it is explained that one may need to set `service_name` to indicate the CI service being used.

But we could not find any place that documented the actual values supported by `service_name`. As with most parts of the API "documentation", only some examples are given, from which we can deduce that "travis-ci" and "travis-pro" are supported. 

Right now, the docs say this:

> Setting `service_name` to "travis-ci" and `service_job_id` to the Travis Job ID will automatically look up the appropriate build and repository, and assign the job correctly on Coveralls.

My best guess is that the documentation really is meant to go on like this:

> Any other values for `service_name` is ignored.

Is that correct? What about the value "travis-pro", which is mentioned elsewhere?

**Comment by fingolfin** [comment]
This is still an issue

**Comment by afinetooth** [comment]
Hi @fingolfin, the specific values for `service_name` depend on what integration you're using and whether that integration supports the particular CI service. 

This is a result of there being many community-created integrations for coveralls, which embodies our original growth strategy to quickly support other languages and CI services beyond (the original) Ruby and Travis. 

We're working on updating our documentation to clarify some topics like this.

In the meantime, the only values for the `service_name` parameter that impact the way the Coveralls API processes incoming requests is:

- **travis-ci** - The original, free Travis CI service (travis-ci.org), now defunct
- **travis-pro** - The new, paid Travis CI service (travis-ci.com), with free tier, now the only Travis option
- **github** - An option for Github Actions CI users, when using `{{ secrets.GITHUB_TOKEN }}` as repo identifier in lieu of the `COVERALLS_REPO _TOKEN`, originally created as the setup for the [Coveralls Github Action](https://github.com/marketplace/actions/coveralls-github-action).

Besides these and their particular use cases, _any_ value can be passed for `service_name`. For instance, for customers using Github Actions CI _without_ the [Coveralls Github Action](https://github.com/marketplace/actions/coveralls-github-action), we recommend using `github-actions` (to distinguish from `github`), which allows the user to continue using the `COVERALLS_REPO_TOKEN` for repo identification.

Basically, any values besides those mentioned above can be used, along with the traditional `COVERALLS_REPO_TOKEN` as repo identifier.

You'll just want to consult your particular language integration to see if a certain value for `service_name` is required to let the integration identify a service it natively supports.

Hope that helps.

If you're trying to dial in `service_name` for your setup, please share details on your CI and the language integration you're using. Generally speaking, those details can be found in the integration's README.
